unblob/handlers/archive/arj.py

```diff
@@ -0,0 +1,184 @@
+import io
+from typing import List, Union
+
+from dissect.cstruct import cstruct
+from structlog import get_logger
+
+from ...file_utils import read_until_null
+from ...models import Handler, UnknownChunk, ValidChunk
+
+logger = get_logger()
+
+
+class ARJNullFileException(Exception):
+    pass
+
+
+class ARJExtendedHeaderException(Exception):
+    pass
+
+
+class ARJFileLengthException(Exception):
+    pass
+
+
+class ARJHandler(Handler):
```

Comment:
Rebase onto main and move to a StructHandler to have access to functions like `parse_header`.